### PR TITLE
Backend/Config: fix config dir not found on macOS

### PR DESCRIPTION
**Pull request by nodefect**
On macOS, NET8 changed Environment.SpecialFolder.ApplicationData
to point to ~/Library/Application Support instead of ~/.config,
which broke config discovery for users who upgraded from a .NET6
build of gwallet to a .NET8 build (their existing config dir at
~/.config/gwallet was no longer found, so a fresh empty config
was created instead).

This change:
- Keeps the old behaviour (based on Env.SF.ApplicationData) for
Windows and mobile (Android/iOS) platforms, where it's still
correct.
- For Unix (macOS/Linux), switches to using ~/.config/gwallet
(via Env.SF.UserProfile), which was the original for those OSs
before the breaking change.
- If the new location doesn't have an existing config yet, it
probes the old possible locations (the pre-.NET8
ApplicationData/gwallet, and the macOS-specific
Library/Application Support/gwallet) and, if found, copies
the whole directory over to the new location, so existing users
don't lose their accounts/config.
- The old directory is only deleted once we have successfully
retrieved balances and confirmed at least one of them is
positive, to minimize risk of data loss in edge cases (e.g. if
the copy went wrong, or if the balances couldn't be fetched).
- Added unit tests covering the migration logic in isolation.

Wrt Essentials, note:
- The modern Backend project now references the DotNetEssentials
nuget package (it didn't ref any Essentials package before).
- The legacy Backend project now references the same
DotNetEssentials version (instead of the Xamarin.Essentials
1.0.0 package), which is the same version that the master branch
references.
- The Device.RuntimePlatform API (and its Device.Android/iOS/
macOS/GTK/UWP/WPF string constants) that IsDesktop() uses in
the master branch does not come from the DotNetEssentials
package, but from the Xamarin.Forms package, which is only
referenced by the GWallet.Frontend.XF project, not by this
Backend library.
- Also, we don't `open` the Xamarin.Essentials namespace and
instead hardcode it in all elements (in the same way as the
master branch's Backend) because opening it fails to compile in
the legacy framework build.

Fixes https://github.com/nblockchain/geewallet/issues/312

### CODE REVIEW

File: src/GWallet.Backend/Config.fs [hunk 1]
```diff
@@ -63,13 +82,111 @@ module Config =
...
+    // still used by Mobile & Windows for now
+    let private GetLegacyConfigDirForThisProgram() =
         let configPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
         let configDir = DirectoryInfo(Path.Combine(configPath, "gwallet"))
-        if not configDir.Exists then
-            configDir.Create()
         configDir
 
+    // used by Unix (macOS & Linux) now
+    let private GetModernConfigDirForThisProgram() =
```
Comment (nodefect) [inline src/GWallet.Backend/Config.fs:92]: Note: on Linux this hardcodes `~/.config`, whereas `ApplicationData` honors `XDG_CONFIG_HOME` for users who set it. Not a problem in practice — the migration probes the legacy dir as a candidate, so a `$XDG_CONFIG_HOME/gwallet` config would get picked up and moved — but it does mean we now permanently diverge from the XDG spec for those users. Fine as a tradeoff, just worth being aware of (and maybe worth a line in the comment if XDG compliance ever matters).

File: src/GWallet.Backend/Config.fs [hunk 2]
```diff
@@ -63,13 +82,111 @@ module Config =
...
+                let configDir = GetModernConfigDirForThisProgram()
+
+                if not (HasExistingConfig configDir) then
+                    let oldConfigDirCandidates = seq {
+                        // the folder that old geewallet versions used on these platforms:
+                        yield GetLegacyConfigDirForThisProgram()
+
+                        // the folder that .NET8+ points at with ApplicationData on macOS:
+                        let configPath = Environment.GetFolderPath Environment.SpecialFolder.UserProfile
+                        yield DirectoryInfo(Path.Combine(configPath, "Library", "Application Support", "gwallet"))
```
Comment (nodefect) [inline src/GWallet.Backend/Config.fs:162]: FYI, on .NET8 macOS these two candidates are the *same* path (`ApplicationData` == `~/Library/Application Support` there), so the seq yields a duplicate — harmless since `Seq.tryHead` takes the first. The hardcoded one is actually load-bearing for the legacy/Mono build on macOS, where `ApplicationData` still maps to `~/.config` (i.e. the new dir, which gets filtered out by the `FullName <>` guard). Might be worth a word in the comment so future readers don't assume one of the two candidates is dead code.

File: src/GWallet.Backend/Config.fs [hunk 3]
```diff
@@ -63,13 +82,111 @@ module Config =
...
+
+                if not configDir.Exists then
+                    configDir.Create()
+                configDir
+            )
+
+    // to be called by frontends when account balances have been retrieved: if a config dir
+    // migration happened, and some retrieved balance is positive, then the new config dir
+    // location is confirmed to be working fine, and the old config dir can be removed
+    let MaybeRemoveMigratedOldConfigDir (): unit =
```
Comment (nodefect) [inline src/GWallet.Backend/Config.fs:174]: One consequence worth spelling out: if the first run after migration ends with all balances zero (or all `Fresh` retrievals failing / only `Cached` values coming back), the old dir is never deleted — and it never will be, because subsequent runs see the new dir's `accounts` dir, skip migration entirely, and `migratedOldConfigDir` stays `None` forever. That's a safe failure mode (leftover dir, no data loss) and I'm fine with it as-is; but if we ever want to close that gap, a small marker file in the new dir (e.g. `migration-pending`) checked at startup would allow retrying the deletion on later runs.

File: src/GWallet.Frontend.Console/UserInteraction.fs [hunk 4]
```diff
@@ -271,6 +271,14 @@ module UserInteraction =
         Console.Write "Retrieving balances..."
         Async.Parallel accountAndBalancesToBeQueried
 
+    let private AnyBalanceIsPositive (accountsWithBalances: seq<IAccount*MaybeCached<decimal>*MaybeCached<decimal>>)
```
Comment (nodefect) [inline src/GWallet.Frontend.Console/UserInteraction.fs:274]: Two small observations:

- Requiring `Fresh` (rather than also accepting `Cached`) is nicely conservative — cached balances could have been migrated along with the config dir itself, so server confirmation is the right signal here. 👍
- This is only wired into the `WhichAccount.All` branch of `DisplayAccountStatuses`; the `MatchingWith` branch (the send flow) also retrieves balances but doesn't call `MaybeRemoveMigratedOldConfigDir`. Harmless in practice since the main status screen runs on every `ProgramMainLoop` iteration — just noting the asymmetry in case you'd rather have it in both branches.
